algorithms/a3c/main.py

```diff
@@ -40,80 +54,275 @@
+parser.add_argument('--interactive', type=int, default=0,
+                    help="""Interactive mode enables human to play
+                    (default: 0)""")
+parser.add_argument('--all-instr-file', type=str,
+                    default="vizdoom_data/instructions_all.json",
+                    help="""All instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_all.json)""")
+parser.add_argument('--train-instr-file', type=str,
+                    default="vizdoom_data/instructions_train.json",
+                    help="""Train instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_train.json)""")
+parser.add_argument('--test-instr-file', type=str,
+                    default="vizdoom_data/instructions_test.json",
+                    help="""Test instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_test.json)""")
+parser.add_argument('--object-size-file', type=str,
+                    default="vizdoom_data/object_sizes.txt",
+                    help='Object size file path relative to a3c folder '
+                         '(default: data/object_sizes.txt)')
+parser.add_argument('-e', '--evaluate', type=int, default=0,
+                    help="""0:Train, 1:Evaluate MultiTask Generalization
+                    2:Evaluate Zero-shot Generalization (default: 0)
+                    async must be on and will only run test function""")
 
 if __name__ == '__main__':
-    os.environ['OMP_NUM_THREADS'] = '1'
     os.environ['CUDA_VISIBLE_DEVICES'] = ""
 
+    # todo add to README.md more info
+
     args = parser.parse_args()
+    # set to 0 so that checkpoint resume can overwrite if necessary
+    args.episode_number = 0
+    args.total_length = 0
+    args.num_backprops = 0
+    if not args.experiment_id:
+        args.experiment_id = datetime.datetime.now().strftime("%Y-%m-%d-") \
+                                                                     + str(uuid.uuid4())
+    args.experiment_path = os.path.join(os.path.abspath(os.path.join(os.path.dirname(__file__),
+                           '..', '..', 'experiments', args.experiment_id)))
+    args.checkpoint_path = os.path.join(args.experiment_path, 'checkpoints')
+    args.tensorboard_path = os.path.join(args.experiment_path, 'tensorboard_logs')
 
-    torch.manual_seed(args.seed)
+    # Create environment from choices of Atari, ViZDoom and our ai2thor wrapper with tasks
     if args.atari:
         env = create_atari_env(args.atari_env_name)
-        args.frame_dim = 42  # fixed to be 42x42 in envs.py _process_frame42()
+        args.resolution = (42, 42)  # fixed to be 42x42 for _process_frame42() in envs.py
+    elif args.vizdoom:
+        # many more dependencies required for VizDoom so therefore import here
+        from algorithms.a3c.env_vizdoom import GroundingEnv
+
+        if args.evaluate == 0:
```

Comment:
I'd rather have strings than 0, 1 and 2. If we start adding this codes then people will have to memorize them to use the code. I'd do:
``` --evaluate "train" ```
And be happier than 0

Reply:
I agree but i might keep it the same due to me not writing the ViZDoom part. If we ever extend the ViZDoom part we can change it all we want

Reply:
Go to slack for discussing this. In general I would say don't touch this ugly code and leave it for reproducibility of the original code. But this hurts deep inside because we are constantly adding and ignoring the fact that we have this weird things in OUR repo

Reply:
From the second we start using vizdoom further from reproducing the original training I would refactor it. For now let's leave it as it is.